cmake/modules/RootBuildOptions.cmake

```diff
@@ -178,6 +178,9 @@ option(rootbench "Build rootbench if rootbench exists in root or if it is a sibl
 option(roottest "Build roottest if roottest exists in root or if it is a sibling directory." OFF)
 option(testing "Enable testing with CTest" OFF)
 
+option(ROOT_PYTHON_NO_LINK "Avoid linking to Python for package managers like Conda" OFF)
```

Comment:
I think that any patch specific to a package manager should live in the respective packaging scripts, etc. That's what I do for Gentoo, for example. See [here](https://gitweb.gentoo.org/repo/gentoo.git/tree/sci-physics/root/files?id=05ef4cae5065cbf82803130b8a74c49795a864a1) for the list of patches I currently have.

Reply:
@henryiii correct me if I’m wrong, but this isn’t specific to conda necessarily, the problem is that the conda ships a statically linked python executable on macOS. Perhaps the comment should be changed to “Avoid linking to Python for statically linked Python installations”?

Reply:
I can change the text. Yes, I believe if you statically link Python when you build Python, it cannot be connected to ROOT without this option. Anaconda Python3 happens to be statically linked.

Reply:
I was referring to the whole thing, not just this particular line. The idea of not linking to the Python library is specific to conda at least for the moment. Since there are the complications when PyROOT is used outside of Python scripts (e.g. when using `TPython` from C++ code), I think we need to let this idea mature a bit more before adding the changes to ROOT proper. Maybe separating the Python module from the parts used by C++ so that one can just be built without linking, but the other can link and work as expected. I'm not against the change, but I think ROOT is just not ready for it yet. I also think that due to ROOT not being ready for it, the changes are a bit hacky (i.e. trying to load `libpythonX.Y` when needed). If the idea is to be able to use ROOT with a different interpreter than it was built for, then we need to not hard-code the library name, which at first impression is locking ROOT to the same Python, just with a different method. For example, the loading of `libpython` will likely fail if Python is statically linked (e.g. you may try to load the system Python instead of the anaconda Python, but the anaconda Python's symbols were copied via statically linking already).

Reply:
Okay, this is something that does need to be addressed but it should be done properly and well. 

The goal is not to connect to a different libPython, though, just a macOS Python 3 that was built with the `--disable-shared` option, which happens to be the way Anaconda builds theirs (due to performance reasons, I believe).

Reply:
Note: Anaconda Python both statically links, and produces the dynamic libpython library too, which might be helping the confusion here. But that's a valid thing to do and ROOT should support it.